helpers/rollout_restart_test.go

```diff
@@ -164,10 +164,13 @@ func assertPatchedRolloutRestartObj(t *testing.T, ctx context.Context, obj ctrlc
 	switch o := obj.(type) {
 	case *appsv1.Deployment:
 		restartAt = o.Spec.Template.ObjectMeta.Annotations[AnnotationRestartedAt]
+		assertFieldManagerSet(t, obj, FieldManagerName)
 	case *appsv1.StatefulSet:
 		restartAt = o.Spec.Template.ObjectMeta.Annotations[AnnotationRestartedAt]
+		assertFieldManagerSet(t, obj, FieldManagerName)
 	case *appsv1.DaemonSet:
 		restartAt = o.Spec.Template.ObjectMeta.Annotations[AnnotationRestartedAt]
+		assertFieldManagerSet(t, obj, FieldManagerName)
```

Comment:
For completeness, I wonder if we should assert that the field manager was not added in the ArgoRollout case below?

Reply:
Thanks for the initial pass @benashz. Actually i discovered the test itself is not testing anything regarding the field manager. Because its passed in as an option to the `client.Patch` function. 

Im not super familiar with the kubernetes client; it seems to be a fake implementation of the actual Kubernetes client. Do you have an example of how to inspect the request potentially to make sure the options are really added to the request? I guess in other apps i would use a mock of the client and expect 

As for the test for making sure its not applied to the argo rollout, makes complete sense. Will do it when i figure out how to actually inspect the options ahaha.